Source/Core/InputCommon/ControlReference/ExpressionParser.cpp

```diff
@@ -288,45 +411,270 @@ class BinaryExpression : public Expression
   }
 };
 
-class UnaryExpression : public Expression
+class FunctionExpression : public Expression
 {
 public:
-  TokenType op;
-  std::unique_ptr<Expression> inner;
+  int CountNumControls() const override
+  {
+    int result = 0;
+
+    for (auto& arg : m_args)
+      result += ' ' + arg->CountNumControls();
```

Comment:
Result isn't a string, did you really intend to add a space (or, `32`) here? If so, could you use the number instead of the character?
Feels a bit like Copy&Paste from `operator std::string()`, so it should probably be just `result += arg->CountNumControls();` instead?

Reply:
Heh. Good catch. Stupid copy pasting got me..